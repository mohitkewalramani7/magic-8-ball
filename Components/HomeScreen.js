/**
 * This script that displays the home screen
 * 
 * @author Mohit Kewalramani
 * @version 1.0
 * @since 2017-08-07
 */

import React from 'react';
import {Image, Text, View, Button} from 'react-native';

import styles from '../Stylesheet';

/**
 * The class that implements the HomeScreen
 */
class HomeScreen extends React.Component {

  // Sets the title of the screen
  static navigationOptions = {
    title: 'Welcome',
  };

  /**
   * The method that renders the home screen's layout
   * and message
   * @return (JSX) The code to render on the screen
   */
  render() {


    // Object of image displayed on the screen
    let ballImage = {
      uri : 'https://www.horoscope.com/images-US/games/game-magic-8-ball-no-text.png'
    };

    return (
      <View style={styles.container}>
        <Text style={styles.title}>An Example Of The Magic 8-Ball</Text>
        <Image source = {ballImage} style = {styles.image}/>
        <Button style={styles.button}
          title="Shake The Ball"
          onPress = {()=> this.props.navigation.navigate('Result')}
        />
        <Button style={styles.button}
          title = "View The Available Options"
          onPress = {()=> this.props.navigation.navigate('Options')}
        />
      </View>
    );
  }
}

// Makes the HomeScreen class public
module.exports = HomeScreen;