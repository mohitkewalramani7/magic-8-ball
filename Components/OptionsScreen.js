/**
 * This script that displays the options list screen.
 * (Showing the user a list of options the 8-ball can 
 * select from)
 * 
 * @author Mohit Kewalramani
 * @version 1.0
 * @since 2017-08-07
 */

import React from 'react';
import {FlatList, Text, View,} from 'react-native';

import styles from '../Stylesheet'

/**
 * The class that implements the OptionsScreen
 */
class OptionsScreen extends React.Component{

  // Sets the title of the screen
  static navigationOptions = {
    title: 'Options for the 8-Ball',
  };

  /**
   * The method that renders the options screen's layout
   * and message
   * @return (JSX) The code to render on the screen
   */
  render(){
    return (
      <View style={styles.container}>
        <Text style={styles.optionsTitle}> Options For The Magic 8 Ball </Text>
        <FlatList
          style = {styles.listview}

          // List of data that is populated on the FlatList (ListView)
          data = {[
              {key: 'It is certain', color : '#b3ffb3'},
              {key: 'It is decidely so', color : '#b3ffb3'},
              {key: 'Without a doubt', color : '#b3ffb3'},
              {key: 'Yes definitely', color : '#b3ffb3'},
              {key: 'You may rely on it', color : '#b3ffb3'},
              {key: 'As I see it, yes', color : '#b3ffb3'},
              {key: 'Most likely', color : '#b3ffb3'},
              {key: 'Outlook good', color : '#b3ffb3'},
              {key: 'Yes', color : '#b3ffb3'},
              {key: 'Signs point to yes', color : '#b3ffb3'},
              {key: 'Reply hazy try again', color : '#ffff99'},
              {key: 'Ask again later', color : '#ffff99'},
              {key: 'Better not tell you now', color : '#ffff99'},
              {key: 'Cannot predict now', color : '#ffff99'},
              {key: 'Concentrate and ask again', color : '#ffff99'},
              {key: 'Don\'t count on it', color : '#ff8080'},
              {key: 'My reply is no', color : '#ff8080'},
              {key: 'My sources say no', color : '#ff8080'},
              {key: 'Outlook not so good', color : '#ff8080'},
              {key: 'Very doubtful', color : '#ff8080'},
          ]}
          // For each item :
          renderItem={({item}) => <Text style={{fontSize : 25, textAlign : 'center', backgroundColor : item.color}}>{item.key}</Text>}
        />
      </View>
    );
  }
}

// Makes the OptionsScreen class public
module.exports = OptionsScreen;