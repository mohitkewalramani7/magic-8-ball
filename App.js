/**
 * This script is the entry point of the app
 * 
 * @author Mohit Kewalramani
 * @version 1.0
 * @since 2017-08-07
 */

import React from 'react';
import {StackNavigator} from 'react-navigation';

import HomeScreen from './Components/HomeScreen';
import OptionsScreen from './Components/OptionsScreen';
import ResultScreen from './Components/ResultScreen';

/**
 * The instance of the StackNavigator which initializes the
 * app with three different screens
 */
const SimpleAppNavigator = StackNavigator({
  Home: {screen : HomeScreen},
  Options: {screen : OptionsScreen},
  Result: {screen : ResultScreen}
});

/**
 * Creates an instance of AppNavigation
 */
const AppNavigation = () => (
  <SimpleAppNavigator />
);

/**
 * A class that renders the created instance of 
 * AppNavigation which starts the app using the screen
 * declared with the Home screen
 */
export default class App extends React.Component{

  /**
   * The render method returns the Home screen as
   * declared by the Navigation instance
   * @return (HTML) The code to render the screen
   */
  render(){
    return(
      <AppNavigation/>
    );
  }
}
