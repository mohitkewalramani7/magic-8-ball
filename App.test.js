import React from 'react';
import App from './App';

import renderer from 'react-test-renderer';
import OptionsScreen from './Components/OptionsScreen';
import HomeScreen from './Components/HomeScreen';

import styles from './Stylesheet'

var constructMessage = require('./Components/ResultScreen').constructmessage;
var ResultScreen = require('./Components/ResultScreen').resultscreen;
var getRandomIndex = require('./Components/ResultScreen').getrandomindex;


describe ("Tests App Without Crashing", () => {
  test("App renders without crashing", () => {
  const rendered = renderer.create(<App />).toJSON();
  expect(rendered).toBeTruthy();
  });
});

describe("Test Individual Screen Components", () => {

  test("Test Options Screen renders correctly", () => {
    const optionsScreenTree = renderer.create(
      <OptionsScreen/>
    ).toJSON();
    expect(optionsScreenTree).toMatchSnapshot();
  });

  test("Home Screen renders correctly", () => {
    const homeScreenTree = renderer.create(
      <HomeScreen/>
    ).toJSON();
    expect(homeScreenTree).toMatchSnapshot();
  });

  test("Test Result Screen renders correctly", () => {
    const resultScreenTree = renderer.create(
      <ResultScreen/>
    ).toJSON();
    expect(resultScreenTree).toMatchSnapshot();
  });

});

describe("Result Screen Helper Methods", () => {

  test("Test Construct Message Length", () => {
    expect(constructMessage().length).toBe(20);
  });

  test("Test Construct Message Content", () => {
    expect(constructMessage()).toContain("It is certain");
    expect(constructMessage()).toContain("It is decidely so");
    expect(constructMessage()).toContain("Without a doubt");
    expect(constructMessage()).toContain("Yes definitely");
    expect(constructMessage()).toContain("You may rely on it");
    expect(constructMessage()).toContain("As I see it, yes");
    expect(constructMessage()).toContain("Most likely");
    expect(constructMessage()).toContain("Outlook good");
    expect(constructMessage()).toContain("Yes");
    expect(constructMessage()).toContain("Signs point to yes");
    expect(constructMessage()).toContain("Reply hazy try again");
    expect(constructMessage()).toContain("Ask again later");
    expect(constructMessage()).toContain("Better not tell you now");
    expect(constructMessage()).toContain("Cannot predict now");
    expect(constructMessage()).toContain("Concentrate and ask again");
    expect(constructMessage()).toContain("Don\"t count on it");
    expect(constructMessage()).toContain("My reply is no");
    expect(constructMessage()).toContain("My sources say no");
    expect(constructMessage()).toContain("Outlook not so good");
    expect(constructMessage()).toContain("Very doubtful");
  });

  test("Test Get Random Index", () => {
    expect(getRandomIndex()).toBeGreaterThanOrEqual(0);
    expect(getRandomIndex()).toBeLessThanOrEqual(19);
  });

});

describe("Test the contents of the Stylesheet", () => {

  test("Test the container properties", () => {
    expect(styles.container["flex"]).toBe(1);
    expect(styles.container["backgroundColor"]).toBe("#fff");
    expect(styles.container["alignItems"]).toBe("center");
    expect(styles.container["justifyContent"]).toBe("center");
  });

  test("Test the title properties", () => {
    expect(styles.title["flex"]).toBe(0.1);
    expect(styles.title["fontSize"]).toBe(35);
    expect(styles.title["textAlign"]).toBe("center");
    expect(styles.title["marginTop"]).toBe(30);
  });

  test("Test the optionsTitle properties", () => {
    expect(styles.optionsTitle["flex"]).toBe(0.3);
    expect(styles.optionsTitle["fontSize"]).toBe(35);
    expect(styles.optionsTitle["textAlign"]).toBe("center");
    expect(styles.optionsTitle["marginTop"]).toBe(35);
  });

  test("Test the button properties", () => {
    expect(styles.button["flex"]).toBe(1);
  });

  test("Test the listview properties", () => {
    expect(styles.listview["flex"]).toBe(1);
    expect(styles.listview["marginTop"]).toBe(25);
    expect(styles.listview["marginBottom"]).toBe(30);
  });

  test("Test the listitem properties", () => {
    expect(styles.listitem["flex"]).toBe(1);
    expect(styles.listitem["fontSize"]).toBe(25);
    expect(styles.listitem["textAlign"]).toBe("center");
  });

  test("Test the image properties", () => {
    expect(styles.image["width"]).toBe(250);
    expect(styles.image["height"]).toBe(250);
    expect(styles.image["alignItems"]).toBe("center");
  });

  test("Test the message properties", () => {
    expect(styles.message["flex"]).toBe(0.5);
    expect(styles.message["marginTop"]).toBe(100);
    expect(styles.message["fontSize"]).toBe(30);
    expect(styles.message["textAlign"]).toBe("center");
  });

});