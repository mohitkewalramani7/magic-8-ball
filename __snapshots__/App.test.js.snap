// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Individual Screen Components Home Screen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "flex": 0.1,
        "fontSize": 35,
        "marginTop": 30,
        "textAlign": "center",
      }
    }
  >
    An Example Of The Magic 8-Ball
  </Text>
  <Image
    source={
      Object {
        "uri": "https://www.horoscope.com/images-US/games/game-magic-8-ball-no-text.png",
      }
    }
    style={
      Object {
        "alignItems": "center",
        "height": 250,
        "width": 250,
      }
    }
  />
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Shake The Ball
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        View The Available Options
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test Individual Screen Components Test Options Screen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "flex": 0.3,
        "fontSize": 35,
        "marginTop": 35,
        "textAlign": "center",
      }
    }
  >
     Options For The Magic 8 Ball 
  </Text>
  <RCTScrollView
    data={
      Array [
        Object {
          "color": "#b3ffb3",
          "key": "It is certain",
        },
        Object {
          "color": "#b3ffb3",
          "key": "It is decidely so",
        },
        Object {
          "color": "#b3ffb3",
          "key": "Without a doubt",
        },
        Object {
          "color": "#b3ffb3",
          "key": "Yes definitely",
        },
        Object {
          "color": "#b3ffb3",
          "key": "You may rely on it",
        },
        Object {
          "color": "#b3ffb3",
          "key": "As I see it, yes",
        },
        Object {
          "color": "#b3ffb3",
          "key": "Most likely",
        },
        Object {
          "color": "#b3ffb3",
          "key": "Outlook good",
        },
        Object {
          "color": "#b3ffb3",
          "key": "Yes",
        },
        Object {
          "color": "#b3ffb3",
          "key": "Signs point to yes",
        },
        Object {
          "color": "#ffff99",
          "key": "Reply hazy try again",
        },
        Object {
          "color": "#ffff99",
          "key": "Ask again later",
        },
        Object {
          "color": "#ffff99",
          "key": "Better not tell you now",
        },
        Object {
          "color": "#ffff99",
          "key": "Cannot predict now",
        },
        Object {
          "color": "#ffff99",
          "key": "Concentrate and ask again",
        },
        Object {
          "color": "#ff8080",
          "key": "Don't count on it",
        },
        Object {
          "color": "#ff8080",
          "key": "My reply is no",
        },
        Object {
          "color": "#ff8080",
          "key": "My sources say no",
        },
        Object {
          "color": "#ff8080",
          "key": "Outlook not so good",
        },
        Object {
          "color": "#ff8080",
          "key": "Very doubtful",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onViewableItemsChanged={undefined}
    renderItem={[Function]}
    renderScrollComponent={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flex": 1,
        "marginBottom": 30,
        "marginTop": 25,
      }
    }
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "#b3ffb3",
              "fontSize": 25,
              "textAlign": "center",
            }
          }
        >
          It is certain
        </Text>
      </View>
      <View
        onLayout={[Function]}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "#b3ffb3",
              "fontSize": 25,
              "textAlign": "center",
            }
          }
        >
          It is decidely so
        </Text>
      </View>
      <View
        onLayout={[Function]}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "#b3ffb3",
              "fontSize": 25,
              "textAlign": "center",
            }
          }
        >
          Without a doubt
        </Text>
      </View>
      <View
        onLayout={[Function]}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "#b3ffb3",
              "fontSize": 25,
              "textAlign": "center",
            }
          }
        >
          Yes definitely
        </Text>
      </View>
      <View
        onLayout={[Function]}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "#b3ffb3",
              "fontSize": 25,
              "textAlign": "center",
            }
          }
        >
          You may rely on it
        </Text>
      </View>
      <View
        onLayout={[Function]}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "#b3ffb3",
              "fontSize": 25,
              "textAlign": "center",
            }
          }
        >
          As I see it, yes
        </Text>
      </View>
      <View
        onLayout={[Function]}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "#b3ffb3",
              "fontSize": 25,
              "textAlign": "center",
            }
          }
        >
          Most likely
        </Text>
      </View>
      <View
        onLayout={[Function]}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "#b3ffb3",
              "fontSize": 25,
              "textAlign": "center",
            }
          }
        >
          Outlook good
        </Text>
      </View>
      <View
        onLayout={[Function]}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "#b3ffb3",
              "fontSize": 25,
              "textAlign": "center",
            }
          }
        >
          Yes
        </Text>
      </View>
      <View
        onLayout={[Function]}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "#b3ffb3",
              "fontSize": 25,
              "textAlign": "center",
            }
          }
        >
          Signs point to yes
        </Text>
      </View>
      <View
        style={
          Object {
            "height": 0,
          }
        }
      />
    </View>
  </RCTScrollView>
</View>
`;

exports[`Test Individual Screen Components Test Result Screen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "flex": 0.1,
        "fontSize": 35,
        "marginTop": 30,
        "textAlign": "center",
      }
    }
  >
    The Magic 8-Ball Says:
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "flex": 0.5,
        "fontSize": 30,
        "marginTop": 100,
        "textAlign": "center",
      }
    }
  >
    Signs point to yes
  </Text>
</View>
`;
